test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpFnTest.java

```diff
@@ -0,0 +1,222 @@
+                                (CUSTOMERS - 1 + FOO_CUSTOMERS) *
+                                        CUSTOMER_ELEMENTS},
+                {"count(//Customer[1]/following-sibling::*)",
+                        CUSTOMERS - 1 + FOO_CUSTOMERS},
+                {"count(//Customer[3]/preceding::*)",
+                        CUSTOMERS - 1 + (CUSTOMERS - 1) * CUSTOMER_ELEMENTS},
+                {"count(//Customer[3]/preceding-sibling::*)", CUSTOMERS - 1},
+                {"count(//Customer[1]/ancestor::*)", ROOT},
+                {"count(//Customer[1]/ancestor-or-self::*)", ROOT + 1},
+                {"count(//Customer[1]/descendant::*)", CUSTOMER_ELEMENTS},
+                {"count(//Customer[1]/descendant-or-self::*)",
+                        CUSTOMER_ELEMENTS + 1},
+                {"count(//Customer/node())",
+                        ID_ATTRIBUTES + CUSTOMERS * CUSTOMER_ELEMENTS},
+        };
+    }
+
+    /*
+     * DataProvider for XPath expression for position function.
+     * Data columns:
+     *  see parameters of the test "testPositionFn"
+     */
+    @DataProvider(name = "positionExpTestCases")
+    public Object[][] getPositionExp() {
+        return new Object[][]{
+                {"//Customer[position()=1]", "Customer_x1"},
+                {"//Customer[position()=last()]", "Customer_x3"},
+                {"//Customer[position()>1 and position()<last()]",
+                        "Customer_x2"},
+                {"//Customer[position() mod 2 =0]", "Customer_x2"},
+                {"//Customer[last()]", "Customer_x3"},
+        };
+    }
+
+    /*
+     * DataProvider for XPath expression for name and local name function.
+     * Data columns:
+     *  see parameters of the test "testNameFn"
+     */
+    @DataProvider(name = "nameExpTestCases")
+    public Object[][] getNameExp() {
+        return new Object[][]{
+                {"local-name(//Customer)", "Customer"},
+                {"local-name(//foo:Customer)", "Customer"},
+                {"local-name(//Customer/@id)", "id"},
+                {"local-name(//foo:Customer/@foo:id)", "id"},
+                {"local-name(//*[local-name()='Customer'])", "Customer"},
+                {"namespace-uri(.)", ""},
+                {"namespace-uri(//Customers)", ""},
+                {"namespace-uri(//Customer)", ""},
+                {"namespace-uri(//foo:Customer)", "foo"},
+                {"namespace-uri(//@id)", ""},
+                {"namespace-uri(//@foo:id)", "foo"},
+                {"name(//*[namespace-uri()=\"foo\"])", "foo:Customer"},
+                {"name(//Customer)", "Customer"},
+                {"name(//foo:Customer)", "foo:Customer"},
+                {"name(//Customer/@id)", "id"},
+                {"name(//foo:Customer/@foo:id)", "foo:id"},
+                {"name(//*[name()='foo:Customer'])", "foo:Customer"},
+        };
+    }
+
+    /**
+     * This test evaluates XPath expressions of id function and checks against
+     * the expected result.
```

Comment:
Use 3rd person descriptive, e.g. "Verifies that the result of evaluating the id function matches the expected result."